Add a Slots library, and a transient variant of ReentrancyGuard (#4955)
OpenZeppelin/openzeppelin-contracts: contracts/utils/README.adoc

 * {MerkleProof}: Functions for verifying https://en.wikipedia.org/wiki/Merkle_tree[Merkle Tree] proofs.
 * {EIP712}: Contract with functions to allow processing signed typed structure data according to https://eips.ethereum.org/EIPS/eip-712[EIP-712].
 * {ReentrancyGuard}: A modifier that can prevent reentrancy during certain functions.

>  * {TransientReentrancyGuard}: Variant of {ReentrancyGuard} that uses transient storage (EIP-1153).

Author: ```suggestion
 * {ReentrancyGuardTransient}: Variant of {ReentrancyGuard} that uses transient storage (EIP-1153).
```

 * {Strings}: Common operations for strings formatting.
 * {ShortString}: Library to encode (and decode) short strings into (or from) a single bytes32 slot for optimizing costs. Short strings are limited to 31 characters.
 * {StorageSlot}: Methods for accessing specific storage slots formatted as common primitive types.

>  * {TransientStorage}: Methods for reading and writing to transient storage.

Author: ```suggestion
 * {Slots}: Methods for derivating, reading and writing to storage slots. This supports both "normal" and transient storage.
```

{{ReentrancyGuard}}

> {{TransientReentrancyGuard}}

Author: ```suggestion
{{ReentrancyGuardTransient}}
```

{{StorageSlot}}

> {{TransientStorage}}

Author: ```suggestion
{{Slots}}
```
